### DISCUSSION TITLE
CSGO server seg faults whenever player joins

### DISCUSSION THREAD

**Discussion by KorinFlakes** [War3Evo/War3Source-EVO#21] [General]
I've spent the better part of 2 days trying to get this to work and I'm pretty sure I have everything in the right places, but whenever someone tries to join the server I get this error:

`[SM] Warning: plugin War3Source.smx is using more than 100000 handles!`
`./srcds_run: line 318: 124565 Segmentation fault      (core dumped) $HL_CMD`

I'm not really sure where to go from here. Any help would be greatly appreciated.

**Comment by War3Evo** [comment]
What are all the plugins you are running other than War3Source.smx?

Are you compiling it using a Sourcemod 1.9 compiler?

Did you go to change the contents in the file War3Source-EVO/addons/sourcemod/scripting/include/GAME_SWITCHER/currentgame.inc in order to compile for your platform?

Did you compile all of these:

```
War3Source.sp
War3Source_0000_SkillsRaceSwapTest_OPT.sp
War3Source_001_UndeadScourge.sp
War3Source_002_HumanAlliance.sp
War3Source_004_NightElf.sp
War3Source_005_BloodMage_TF2.sp
War3Source_009_CorruptedDisciple.sp
War3Source_010_SoulReaper.sp
War3Source_011_BloodHunter.sp
War3Source_012_Naix.sp
War3Source_013_SuccubusHunter.sp
War3Source_014_Chronos.sp
War3Source_015_Lich.sp
War3Source_016_SacredWarrior.sp
War3Source_017_Hammerstorm.sp
War3Source_018_Scout.sp
War3Source_019_DarkElf.sp
War3Source_020_Dragonborn.sp
War3Source_021_Fluttershy.sp
War3Source_022_Rarity.sp
War3Source_023_RainbowDash.sp
War3Source_028_Luna.sp
War3Source_051_Frogger.sp
War3Source_057_LightBender.sp
War3Source_064_ShadowPaladin.sp
War3Source_066_Soul_Medic_TF2.sp
War3Source_Addon_AdminConsole.sp
War3Source_Addon_AdminMenu.sp
War3Source_Addon_Auction_House.sp
War3Source_Addon_Client_AimTarget_Information.sp
War3Source_Addon_Client_Preferences.sp
War3Source_Addon_Full_Player_Information.sp
War3Source_Addon_Hud_Info.sp
War3Source_Addon_LevelUpParticle.sp
War3Source_Addon_ShopItems.sp
War3Source_Addon_ShopItems2.sp
tf2attributes.sp 
```

and put the resulting SMXs into your game directory along with War3Source.smx?

Anything else did you do?

What does the structure of your directory look like with War3Source.smx installed? (what other files)

**Comment by War3Evo** [comment]
Do note that not everything will compile because a lot of the races haven't been updated for CSGO.

**Comment by geoka99** [comment]
Hi! 
I discovered another problem: `// C:\Users\George\Desktop\Compiler\addons\sourcemod\scripting\War3Source_0000_SkillsRaceSwapTest_OPT.sp(6) : error 001: expected token: "#endif", but found "-end of file-"
//
// 1 Error.
//`
I try with  "#if defined" but...

**Comment by War3Evo** [comment]
This means, if game type is css or CSGO, DON'T COMPILE

```
#if (GGAMETYPE == GGAME_CSS || GGAMETYPE == GGAME_CSGO)
	#endinput
#endif
```

I usually compile everything and when it errors like that, I know it didn't compile that ON PURPOSE.

**Comment by War3Evo** [comment]
The code is written in SourceMod 1.7 & 1.8 language.  Any updates to thier new language won't work.

**Comment by KorinFlakes** [comment]
Sorry for the late reply, I had missed a number of SP files when I compiled it and that's what was causing the crashes.

I've added them all and the server runs fine but after 5 mins of being connected to the server each client gets 
`DispatchAsyncEvent backlog, failed to dispatch all this frame.`

Spammed in their console and your FPS drops drastically. It doesn't happen to each client at the same time, just randomly after a few minutes of being connected. The only way to fix the lag is to restart the entire game once it starts.

The only sourcemod mods I'm running are this one and deathmatch and it only started after I reinstalled the warcraft mod.

**Reply by War3Evo** [reply]
> Sorry for the late reply, I had missed a number of SP files when I compiled it and that's what was causing the crashes.
> 
> I've added them all and the server runs fine but after 5 mins of being connected to the server each client gets `DispatchAsyncEvent backlog, failed to dispatch all this frame.`
> 
> Spammed in their console and your FPS drops drastically. It doesn't happen to each client at the same time, just randomly after a few minutes of being connected. The only way to fix the lag is to restart the entire game once it starts.
> 
> The only sourcemod mods I'm running are this one and deathmatch and it only started after I reinstalled the warcraft mod.

I've never had this issue.

What are your VPS stats?  Are you running at least 1GB of RAM?

I've not had that issue before.

How many clients need to join to start that issue?

You do know you can run the latest SourceMod, but you must compile the code using SourceMod 1.8 maximum.  If your not running the latest SourceMod in CSGO, then do that!  Get the latest SourceMod and MetaMod versions.

**Reply by War3Evo** [reply]
> I've added them all and the server runs fine but after 5 mins of being connected to the server each client gets `DispatchAsyncEvent backlog, failed to dispatch all this frame.`
> 
> Spammed in their console and your FPS drops drastically. It doesn't happen to each client at the same time, just randomly after a few minutes of being connected. The only way to fix the lag is to restart the entire game once it starts.
> 
> The only sourcemod mods I'm running are this one and deathmatch and it only started after I reinstalled the warcraft mod.

This issue has been reported to the Value Servers.  It's a CSGO Death Match game mode issue and not a War3Soruce Issue:
https://github.com/ValveSoftware/csgo-osx-linux/issues/2483

**Comment by War3Evo** [comment]
Go to the link below for a fully (SMX files only) compiled zip and a test server link:

[https://forums.alliedmods.net/showpost.php?p=2798043&postcount=92](https://forums.alliedmods.net/showpost.php?p=2798043&postcount=92)

You'll still need config, cfg, translations, etc.

**Reply by KorinFlakes** [reply]
So I uninstalled the entire csgo server and reinstalled it using just MM source 1.11, Sourcemod 1.11 and the warcraft mod (using the plugins you posted above) and the other files from the github (including the sound files). The server is good for about 10 minutes but then the frame drops begin again. It happens to everyone who joins the server, regardless of whether we play Casual or Deathmatch and regardless of what OS the player is on.

Just for good measure I deleted the W3Source plugins from the plugins folder and restarted the server and the issue went away. Played fine for about 30 mins before I stopped the server again. Fwiw I'm not sure if this matters but the server is running from Pterodactyl-panel.

**Reply by War3Evo** [reply]
My server is actually running on 2GB of RAM for CSGO.  I messed up saying it was 1GB.

Does the server crash after 10 minutes or does the client crash?

If anything, it could be a race causing the issue.  I couldn't figure out which one yet, but I think NightElf is a okay race as it was one of them I updated for CSGO.

Try removing all race files except for War3Source_004_NightElf.

Races are War3Source_000_name pattern.

You must have at least 1 race running to run War3Source.

**Reply by War3Evo** [reply]
Also, try only running on map de_dust2.

**Reply by War3Evo** [reply]
The reason I said to run only Night Elf is because I played night elf and it worked fine for me.  It sends known graphics that works for CSGO.  Although, some computers may not be able to handle War3Source graphics.  War3Source graphics can get intense and older computers may struggle to run it.

In order to use night elf abilities you'll have to bind keys to the abilities and ultimate.  Do you know how to bind keys?

**Reply by War3Evo** [reply]
I tested your theory... War3Source disabled ... disables the problem...

So... I'll add in multiple features... that will "dumb down" War3Source spam.

MY CURRENT THEORY:    WAR3SOURCE by nature is TEXT SPAMMY.   I think it's TEXT spamming and the clients can't handle the TEXT spam... TF2 could handle text spam, but for some reason.... nothing to do with GRAPHICS..lol

I'll test my theory by adding a feature that disables TEXT spam OR reduces it a lot.

**Comment by War3Evo** [comment]
Here's some information on the issue:
https://github.com/ValveSoftware/csgo-osx-linux/issues/2826

**Comment by War3Evo** [comment]
It's fixed in the new War3Source source code.

I'll upload a compiled copy to Allied Modders soon.


The ISSSUE?

I removed FCVAR_REPLICATED and other  ConVars from War3Source.   The solution was found here: [https://github.com/ValveSoftware/csgo-osx-linux/issues/2826#issue-1035632585](https://github.com/ValveSoftware/csgo-osx-linux/issues/2826#issue-1035632585)

**Reply by War3Evo** [reply]
Compiled Source for CSGO: https://forums.alliedmods.net/showpost.php?p=2798088&postcount=94

**Reply by geoka99** [reply]
I tried it and now it works on my server.

**Comment by geoka99** [comment]
![Imagine WhatsApp 2023-01-29 la 15 38 43](https://user-images.githubusercontent.com/123424258/215330130-31a46616-148c-4397-9549-a293145916ce.jpg)

Hi!
Apparently it is the same problem, the fps being very low.

**Comment by War3Evo** [comment]
For the numerous issues that CSGO has given it and my time. I'll just remove it from the TAGs on Allied Modders.

The CSGO version will not be ripped out and someone else wants to solve the issue, then all the source code is here.

I will no longer be supporting CSGO as it's beyond my knowledge to fix it properly.  I also don't play CSGO and have no interest in keeping it up to date.

I will be abandoning CSGO version, however I will continue to support CSS, TF2, and FOF.

Sorry, it just could never work for CSGO.
